src/main/java/org/csanchez/jenkins/plugins/kubernetes/pod/retention/Reaper.java

```diff
@@ -90,10 +98,12 @@ public static Reaper getInstance() {
 
     private Watch watch;
 
+    private final Map<String, Set<String>> terminationReasons = new HashMap<>();
```

Comment:
Does it matter that no entry will ever be removed from this map? Or should additions be infrequent?

Reply:
I think additions should be infrequent enough that we do not really care about slowly growing the map.

Reply:
Couldn't this possibly grow pretty quickly if there is a Pod template fails to launch for a prolonged period of time? For example an Agent is launched, it fails to init due to ImageBackoff, Pod is killed because it didn't start in time, a new Agent is launched, and the cycle continues until someone notices or the job times out. I can attest to running in to this and similar scenarios quite regularly unfortunately.

Is there a way to detect when a node is removed so the associated reasons can be dropped? Or maybe switch to a time/size based cache?

Reply:
> detect when a node is removed so the associated reasons can be dropped

No, in fact the termination reason is commonly needed after the node has been removed due to the pod being deleted.

> switch to a time/size based cache

Yes this would make sense. I think in practice an entry would be used, if at all, within a few minutes, so we could probably delete them after an hour or so.